src/commands/purge.js

```diff
@@ -0,0 +1,159 @@
+const Discord = require('discord.js');
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const util = require('../util');
+
+/**
+ *
+ * @param {Discord.CommandInteraction} interaction
+ */
+async function purgeHandler(interaction) {
+	await interaction.deferReply({
+		ephemeral: true
+	});
+
+	const guild = await interaction.guild.fetch();
+	const executingMember = await interaction.member.fetch();
+	const executor = executingMember.user;
+	const deleteAmount = interaction.options.getInteger('amount');
+	const users = interaction.options.getString('users');
+
+	const eventLogEmbed = new Discord.MessageEmbed();
+
+	const purgeReplyEmbed = new Discord.MessageEmbed();
+	const image = new Discord.MessageAttachment('./src/images/mod/mod-purge.png');
+
+	if (users !== null) { // if a user(s) is actually given
+		purgeReplyEmbed.setTitle('Messages Purged');
+		purgeReplyEmbed.setColor(0xff6363);
+		purgeReplyEmbed.setThumbnail('attachment://mod-purge.png');
+
+		const userIds = [...new Set(Array.from(users.matchAll(Discord.MessageMentions.USERS_PATTERN), match => match[1]))];
+
+		for (const userId of userIds) {
+			const member = await interaction.guild.members.fetch(userId);
+			const user = member.user;
+
+			// Grab the user's messages
+			const userMessages = (await interaction.channel.messages.fetch()).filter(
+				(m) => m.author.id === member.id
+			);
```

Comment:
This is flawed. At most this will return 100 messages, and may miss messages

Take for example if you wanted to purge 20 messages from the user. This function here grabs the last 100 messages in the channel, but those 100 messages may not include the last 20 the user sent

I'm not aware if Discord.JS can do this out of the box, but maybe using the undocumented API endpoint would work? Discord.JS provides a way to make direct API requests, and the search bar uses https://discord.com/api/v9/guilds/GID/messages/search for this

A query to https://discord.com/api/v9/guilds/GID/messages/search?author_id=ID&include_nsfw=true will return specifically that users messages in a JSON response like:

```json
{
    "analytics_id": "some ID",
    "doing_deep_historical_index": false,
    "total_results": 12345,
    "messages": [
        [
            {
                // Messaage data here
            }
        ]
    ]
}
```

Adding an `offset` param gets other pages, since not all messages are returned at once

I'm unsure if using the private API is safe usage however. @DaniElectra @gitlimes @hauntii thoughts?